### PR TITLE
Jupyter (formerly iPython) not needed in setup.py

### LINKED ISSUE #810: Ipython not actually needed for setup
Right now we have an unofficial dependency which is IPython, not listed at www.pyne.io. My guess is is that its actually needed for tests or building documentation? We should therefore move the check in https://github.com/pyne/pyne/blob/develop/setup.py#L153 to a more appropriate place?

### PR DESCRIPTION
**Pull request by opotowsky**
This PR resolves issue #810 by deleting the ipython check from setup.py. Since it is an optional dependency, it has been added as such to the appropriate documentation.

### CODE REVIEW

File: docs/devsguide/testing.rst [hunk 1]
```diff
@@ -32,12 +32,13 @@ passing.
...
-PyNE examples are either in Python files or Jupyter notebooks. This means that
-to test the examples requires a recent version of Jupyter.  Furthermore, the 
-examples themseleves may have many other optional dependencies.  Don't be alarmed
-if testing the examples fails do to a lack of having a dependency installed.
-For this reason, testing the examples is not as important as unit tests, but still
-should be done occassionally.
+PyNE examples are either in Python files or `Jupyter notebooks
+<https://jupyter.org/>`_. This means that to test the examples requires a
+recent version of Jupyter.  Furthermore, the examples themseleves may have many
+other optional dependencies.  Don't be alarmed if testing the examples fails do
```
Comment (nsryan2) [inline docs/devsguide/testing.rst]:
```suggestion
other optional dependencies.  Don't be alarmed if testing the examples fails due
```

Comment (opotowsky) [reply docs/devsguide/testing.rst]: Wow I wonder how long that typo has lingered. Thanks for catching!

File: docs/install/linux_source.rst [hunk 2]
```diff
@@ -20,11 +20,15 @@ PyNE has the following dependencies:
    #. `Python 2.7 <http://www.python.org/>`_
```
Comment (nsryan2) [inline docs/install/linux_source.rst]:
```suggestion
   #. `Python <http://www.python.org/>`_
```

**Comment by nsryan2** [comment]
It looks good to me, though I can't figure out why it isn't passing two of the tests. Are they even important for this PR, because it only has to do with the documentation and examples and not pyne itself?

**Comment by opotowsky** [comment]
It's not passing because of changes to `setup.py`...I'm looking into it

**Comment by opotowsky** [comment]
I think they are red x's b/c they were cancelled builds as I pushed new commits -- looking good so far this time!

**Review by gonuke — APPROVED** [review]
Thanks @opotowsky - I'll wait for testing and then merge

**Pull request merged**